### ISSUE TITLE
add schema to subquery falsely when using withSchema

### ISSUE THREAD

**Issue by xiechao06** [knex/knex#2607]
# Environment

Knex version:
Database + version: postgres 0.14.4
OS: ubuntu 16.04

Select applicable tempalate from below.
If issue is about oracledb support tag @ atiertant. For MSSql tag @ smorey2 .
Rest of dialects doesn't need tags.

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

given the following code:

```javascript
db('departments').withSchema('foo')
  .join(
    db('trainees').withSchema('foo').groupBy('department_id')
      .select('department_id', db.raw('count(*)'))
      .as('trainee_cnts'),
    'trainee_cnts.department_id', 'departments.id'
  )
  .select('departments.*', 'trainee_cnts.count as trainee_cnt')
  .then(console.log);
```
knex add schema 'foo' to subquery and produce the following query:

```
select "departments".*, "trainee_cnts"."count" as "trainee_cnt" from "MY_SCHEMA"."departments" inner join "MY_SCHEMA"."select ""department_id"", count(*) from ""MY_SCHEMA"""."""trainees"" group by ""department_id""" on "trainee_cnts"."department_id" = "departments"."id"'
```

2. Error message
```
Unhandled rejection error: cross-database references are not implemented: "MY_SCHEMA.select "department_id", count(*) from "MY_SCHEMA"."trainees" group by "department_id""
```

**Comment by elhigu** [comment]
Verified that withSchema really causes some kind of string conversion to the subquery. Very strange.

**Comment by DataTables** [comment]
Just to follow up on this - I'm seeing this error as well. Made a short demonstration here: https://runkit.com/datatables/5ef516a9e73fab0019d3f74d .

**Comment by kibertoad** [comment]
Released in 0.95.0
